include/Node/Node.hpp

```diff
@@ -39,23 +39,24 @@ class Node {
   void setId(const std::string &);
 
  public:
-  Node(const std::string &, const T &data);
+  Node(const std::string &, T data);
```

Comment:
It may be worth declaring a move constructor, so that other users of `Node` know that ownership of the data is explicitly being taken.

Reply:
I agree, I'll add it right now